laba4.py

```diff
@@ -0,0 +1,64 @@
+class GemStone:
+    total_gems = 0  
+    store_name = "Luxury Gems" 
+    
+    def __init__(self, carats=None, price=None, name=None, color="Без кольору", origin="Невідоме походження"):
+        self.__carats = carats if carats else 0.0
+        self.__price = price if price else 0.0
+        self.__name = name if name else "Unknown"
```

Comment:
In line 5, you can replace None with the appropriate values to make the code easier to read.